### PR TITLE
Give the API shim the ability to create/update secrets

### PR DESCRIPTION
**Pull request by tcdowney**
## What is this change about?
Give the API shim the ability to create/update secrets so that it is able to create/update app env vars.

## Tag your pair, your PM, and/or team
@matt-royal @akrishna90 @Birdrock

### CODE REVIEW

**Review by gnovv — APPROVED** [review]
lgtm

File: api/config/base/rbac/role.yaml [hunk 1]
```diff
@@ -23,6 +23,9 @@ rules:
   - get
   - list
   - watch
+  - create
```
Comment (tcdowney) [inline api/config/base/rbac/role.yaml:26]: Are these generated somewhere else?

Comment (tcdowney) [reply api/config/base/rbac/role.yaml:26]: I just realized when I run `make build-reference-api` it deletes them.

**Review by akrishna90 — CHANGES_REQUESTED** [review]
@tcdowney - the file you have edited is autogenerated, so these changes are not preserved. 
Please add them as kubebuilder annotations

File: api/config/base/rbac/role.yaml (hunk 1 above)
Comment (akrishna90) [reply api/config/base/rbac/role.yaml:26]: Add them as kubebuilder annotations, you'll be fine

Comment (tcdowney) [reply api/config/base/rbac/role.yaml]: Thanks I added them to the apps repo and pushed a new commit! 👍

**Review by akrishna90 — APPROVED** [review]

**Pull request merged**